debug_toolbar/apps.py

```diff
@@ -1,16 +1,46 @@
 from __future__ import absolute_import, unicode_literals
 
+import warnings
+
 from django.apps import AppConfig
+from django.conf import settings
+from django.middleware.gzip import GZipMiddleware
+from django.utils.module_loading import import_string
 from django.utils.translation import ugettext_lazy as _
 
-from debug_toolbar import settings as dt_settings
+from debug_toolbar.middleware import DebugToolbarMiddleware
 
 
 class DebugToolbarConfig(AppConfig):
     name = 'debug_toolbar'
     verbose_name = _("Debug Toolbar")
 
     def ready(self):
-        if dt_settings.get_patch_settings():
-            dt_settings.patch_all()
-        dt_settings.check_middleware()
+        check_middleware()
+
+
+def check_middleware():
+    gzip_index = None
+    debug_toolbar_index = None
+
+    # Determine the indexes which gzip and/or the toolbar are installed at
+    for i, middleware in enumerate(settings.MIDDLEWARE_CLASSES):
```

Comment:
> IMO the check could also be replaced by a big warning in the installation docs, but not sure whether that would be alright with everyone.

This already exists to some extent. From the [installation docs Middleware section](https://raw.githubusercontent.com/jazzband/django-debug-toolbar/master/docs/installation.rst):

> The order of ``MIDDLEWARE_CLASSES`` is important. You should include the Debug
Toolbar middleware as early as possible in the list. However, it must come
after any other middleware that encodes the response's content, such as
:class:`~django.middleware.gzip.GZipMiddleware`.

Is this sufficient? Or would you like me to expand on this blurb? Or should I continue with modifying the runtime check?